crates/sui-rest-api/src/client/sdk.rs

```diff
@@ -400,30 +404,72 @@ impl Client {
         self.bcs(response).await
     }
 
-    fn check_response(
+    pub async fn simulate_transaction(
+        &self,
+        transaction: &Transaction,
+    ) -> Result<Response<TransactionSimulationResponse>> {
+        let url = self.url().join("transactions/simulate")?;
+
+        let body = bcs::to_bytes(transaction)?;
+
+        let response = self
+            .inner
+            .post(url)
+            .header(reqwest::header::ACCEPT, crate::APPLICATION_BCS)
+            .header(reqwest::header::CONTENT_TYPE, crate::APPLICATION_BCS)
+            .body(body)
+            .send()
+            .await?;
+
+        self.bcs(response).await
+    }
+
+    pub async fn resolve_transaction(
+        &self,
+        unresolved_transaction: &UnresolvedTransaction,
+    ) -> Result<Response<ResolveTransactionResponse>> {
+        let url = self.url.join("transactions/resolve")?;
+
+        let response = self
+            .inner
+            .post(url)
+            .header(reqwest::header::ACCEPT, crate::APPLICATION_BCS)
```

Comment:
Does this API always return BCS, or will it return JSON if you provide an application/json accept header?

Reply:
Intent would be to support BCS or JSON return depending on the accept header